### ISSUE TITLE
Improve event view, filters

### ISSUE THREAD

**Issue by peintnermax** [zitadel/zitadel#5136]
follow up of #4907

As an administrator I want to have an improved view on the events and have a better experience filtering for events

**Acceptance Criteria**
- [ ] filter for event types
- [ ] filter for usernames instead of userId
- [ ] show user avatar, loginnames

**Comment by hifabienne** [comment]
I think the first two options are good. an avatar is from my perspective not needed in a technical view.

**Comment by juergrinaldi** [comment]
Here's a quick and (very) dirty mockup of a possible filter solution as discussed with @peintnermax today

![image](https://user-images.githubusercontent.com/48990649/215790429-42133040-b011-4b78-be3b-37be86d4d4c9.png)

quick explaination:
we implement a filter button to add a new filter
on click the first of these 3 popups shows and shows all possible filter types
user selects one of them and clicks `next`* -> middle pop up shows options according to selected filter type
if needed there will be other input screens like a value input or a selection of possible values
on finish you get a new filter that shows as a bullet like two examples next to the filter buttons, where you can 'x-out' if not needed anymore.

(* maybe it makes sense to get rid of the `next` button and go to the follow up screen on selection. (but keep back and finish buttons))

**Comment by juergrinaldi** [comment]
> I think the first two options are good. an avatar is from my perspective not needed in a technical view.

Can you please elaborate the user story behind this reasoning? I don't understand why it's not helpful for the admin to have additional identifier like profile picture or login names.

What is a practical use case for this issue?
What problem does the user (admin) want to solve with this?

I'm just trying to get my head around this. ;)

**Comment by hifabienne** [comment]
From my perspective this is just a really technical view.
So If something doesn't work I can have a look at the events. 
The API is used so I can get the events to another system and react to them. 
You can find some example use cases in this guide: https://zitadel.com/docs/guides/integrate/event-api
The main use case is probably the API itself, but we just like to provide the view for it so the admins can have a quick look at what the "technical" side looks.
So I don't see a need of an avatar in a technical view.

**Comment by peintnermax** [comment]
I opt for every point which is adding recognisability to the UI. And since we are also showing avatars for system and login in our changed lists, we should also add them to the view above. And Avatars don't take up a lot of space at all, and are easy to implement.
<img width="401" alt="Bildschirm­foto 2023-02-06 um 08 53 53" src="https://user-images.githubusercontent.com/10165752/216915415-1f6b8a4f-263d-4086-8ded-8906b9fd038b.png">

**Comment by peintnermax** [comment]
And at a later point I'd improve the changes response too such that it uses our proto definition of [Editor](https://zitadel.com/docs/apis/proto/event#editor)

**Comment by hifabienne** [comment]
From my perspective It is just something we add that is not used for the moment and has to be maintained. 
A not existing avatar will probably not make the ui ugly and unusable, if that is the case we have a complete different understanding of unusable and ugly. :)

**Comment by juergrinaldi** [comment]
it's not about ugly and unusable (or not exclusive ;)) but it functions as an additional identifier for the user. i.e.: admin searches for users that made changes to Project Z. They get a list of 20 different users, but with two different John Doe. The avatar helps them to identify who's who.

Besides that it is easier to to capture a certain person without having to read names as they eyes are faster in recognizing (visual) patterns than processing text. 
And that it is a technical view (or not) has nothing to do with that. (except API maybe, but that's whole different story and has nothing do to with the UI (as far as I understand that).

TLDR: it allows faster search for users.

**Comment by peintnermax** [comment]
I agree with @juergrinaldi in that case.. I don't get why we are discussing this as implementing this is hardly any effort and goes hand in hand with a good api design. Why don't expose data we have anyways.. More data could also interest api users

**Comment by hifabienne** [comment]
From an event perspective we do not have this data anyways. Every additional data has to be included additionally as I know.

**Comment by hifabienne** [comment]
I will close this issue as the events are more needed by api from customers.
